core/src/main/scala/akka/persistence/jdbc/schema/scaladsl/SchemaUtils.scala

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2014 - 2019 Dennis Vriend <https://github.com/dnvriend>
+ * Copyright (C) 2019 - 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.jdbc.schema.scaladsl
+
+import java.sql.SQLSyntaxErrorException
+import java.sql.Statement
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.dispatch.Dispatchers
+import akka.persistence.jdbc.db.SlickDatabase
+import akka.persistence.jdbc.db.SlickExtension
+import org.slf4j.LoggerFactory
+import slick.jdbc.H2Profile
+import slick.jdbc.JdbcBackend.Database
+import slick.jdbc.MySQLProfile
+import slick.jdbc.OracleProfile
+import slick.jdbc.PostgresProfile
+import slick.jdbc.SQLServerProfile
+
+object SchemaUtils {
+
+  private val logger = LoggerFactory.getLogger("akka.persistence.jdbc.schema.SchemaUtils")
+
+  /**
+   * For testing purposes the journal and snapshot tables can be created programmatically.
+   * For production it's recommended to create the table with DDL statements before the system is started.
+   * For information about the different schemas and supported database can be found at
+   * https://doc.akka.io/docs/akka-persistence-jdbc/current/index.html#database-schema
+   *
+   * This method will automatically detect the configured database using the settings from `jdbc-journal` config.
+   * If configured with `use-shared-db`, it will use the `akka-persistence-jdbc.shared-databases` definition instead.
+   * See https://doc.akka.io/docs/akka-persistence-jdbc/current/index.html#sharing-the-database-connection-pool-between-the-journals for details.
+   *
+   * @param systemProvider a ClassicActorSystemProvider
+   */
+  def dropAndCreate(implicit systemProvider: ClassicActorSystemProvider): Future[Done] = {
```

Comment:
should it be `dropAndCreate()(implicits)` given that it is not a pure method, so that usage becomes `dropAndCreate()`?

Reply:
why two so different names, `dropAndCreate` vs `applySchema` ?

Reply:
> why two so different names, dropAndCreate vs applySchema ?

Because the default schema is a drop and create. 

The `applySchema` can be any DDL script.

Maybe better if we call if `applyDefaultSqlScript` and `applySqlScript`. 

The passed script can be anything you want. It can be an alter table, a drop, inserts.